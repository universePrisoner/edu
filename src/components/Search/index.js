import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './index.css';

class Search extends Component {

    componentDidMount() {
        if (this.input) this.input.focus();
    }

    render() {
        const {
            value,
            onChange,
            onSubmit,
            children
        } = this.props;

        const search = (
            <form className="form" onSubmit={onSubmit}>
                {children} <input ref={(node) => { this.input = node; }} value={value} type="text" className="form__search" onChange={onChange} />
                <button type="submit">{children}</button>
            </form>
        );

        return (search);
    }
}

Search.propTypes = {
    className: PropTypes.string,
    onSubmit: PropTypes.func,
    value: PropTypes.string,
    onChange: PropTypes.func
}

export default Search;