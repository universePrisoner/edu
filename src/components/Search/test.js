import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Search from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Search', () => {
    let properties = {
        className: 'str',
        onSubmit: () => console.log(1),
        value: 'str',
        onChange: () => console.log(2),
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Search  {...properties}>Search</Search>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('here is short snapshot', () => {
        const component = renderer.create(<Search {...properties}>Search</Search>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('show search input', () => {
        let element = shallow(<Search  {...properties}></Search>);

        expect(element.find('.form').length).toBe(1);
    });

});
