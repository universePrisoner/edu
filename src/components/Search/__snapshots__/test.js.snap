// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search here is short snapshot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  Search
   
  <input
    className="form__search"
    onChange={[Function]}
    type="text"
    value="str"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
